src/main/java/com/ignek/crud/servlet/AddUserServlet.java

```diff
@@ -1,81 +1,69 @@
 
-/**
- * Servlet implementation class AddUserServlet
- */
 @WebServlet("/AddUserServlet")
 public class AddUserServlet extends HttpServlet {
 	private static final long serialVersionUID = 1L;
 
-	/**
-	 * @see HttpServlet#HttpServlet()
-	 */
-	public AddUserServlet() {
-		super();
-		// TODO Auto-generated constructor stub
-	}
-
-	/**
-	 * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse
-	 *      response)
-	 */
-	protected void doGet(HttpServletRequest request, HttpServletResponse response)
-			throws ServletException, IOException {
-		
-		response.getWriter().append("Served at: ").append(request.getContextPath());
-	}
-
-	/**
-	 * @see HttpServlet#doPost(HttpServletRequest request, HttpServletResponse
-	 *      response)
-	 */
 	protected void doPost(HttpServletRequest request, HttpServletResponse response)
 			throws ServletException, IOException {
-		
-		String firstname = request.getParameter("firstname"); 
-        String lastname = request.getParameter("lastname");
-        String gender = request.getParameter("gender");
-        String phonenumber = request.getParameter("phonenumber");
-        String emailaddress = request.getParameter("emailaddress");
-        String course = request.getParameter("course");
-        String subjects = request.getParameter("subjects");
-        
-        
-        System.out.println("firstname=====>" + firstname);
-        System.out.println("lastname=====>" + lastname);
-        System.out.println("gender=====>" + gender);
-        System.out.println("phonenumber=====>" + phonenumber);
-        System.out.println("emailaddress=====>" + emailaddress);
-        System.out.println("course=====>" + course);
-        System.out.println("subjects=====>" + subjects);
-        
-        
-        User user = new User();
-        
-        user.setFirstName(firstname);
-        user.setLastName(lastname);
-        user.setGender(gender);
-        user.setPhoneNumber(phonenumber);
-        user.setEmailAddress(emailaddress);
-        user.setCourse(course);
-        user.setSubjects(subjects);
-        
-       int result =  UserInsert.insertUser(user);
-		if (result > 0) {
-			System.out.println("Successfully done");
+		try {
+
+			String userIdString = request.getParameter(UserConstant.USER_ID);
+			String firstname = request.getParameter(UserConstant.FIRST_NAME);
+			String lastname = request.getParameter(UserConstant.LAST_NAME);
+			String gender = request.getParameter(UserConstant.GENDER);
+			String phonenumber = request.getParameter(UserConstant.PHONE_NUMBER);
+			String emailaddress = request.getParameter(UserConstant.EMAIL_ADDRESS);
+			String course = request.getParameter(UserConstant.COURSE);
+
+			String[] subjectsArray = request.getParameterValues(UserConstant.SUBJECTS);
+			String subjects = String.join(",", subjectsArray);
+
+			int userId = userIdString != null && !userIdString.isEmpty() ? Integer.parseInt(userIdString) : 0;
+
+			User user = new User();
```

Comment:
Here you can directly use the below method to create a new DTO with data.

public User(int userId, String firstName, String lastName, String gender, String emailAddress, String phoneNumber,
			String course, String subjects) {}